CMakeLists.txt

```diff
@@ -0,0 +1,377 @@
+  c = strerror_r(0,c,0);
+  return 0;
+}" STRERROR_R_CHAR_P )
+
+check_c_source_compiles ("int main() {
+  _Bool b = 1;
+  return 0;
+}" HAVE__BOOL )
+
+check_c_source_compiles ("#include <sys/types.h>
+int main() {
+  size_t b = 1;
+  return 0;
+}" HAVE_SIZE_T )
+
+if (NOT HAVE_SIZE_T)
+ set (size_t "unsigned int")
+endif ()
+
+check_c_source_compiles ("#include <sys/types.h>
+int main() {
+  ssize_t b = 1;
+  return 0;
+}" HAVE_SSIZE_T )
+
+if (NOT HAVE_SSIZE_T)
+ set (ssize_t "int")
+endif ()
+
+check_c_source_compiles ("#include <assert.h>
+static_assert(sizeof(int) == 4, \"Code relies on int being exactly 4 bytes\");
+int main(void) {
+    return 0;
+}" HAVE_C_STATIC_ASSERT)
+
+if (NOT HAVE_C_STATIC_ASSERT)
+  set ("static_assert" "typedef void _no_static_assert")
+endif ()
+
+foreach (KEYWORD "inline" "__inline__" "__inline")
+  if (NOT DEFINED C_INLINE)
+    set (CMAKE_REQUIRED_DEFINITIONS "-Dinline=${KEYWORD}")
+
+    check_c_source_compiles ("typedef int foo_t;
+      static inline foo_t static_foo(){return 0;}
+      foo_t foo(){return 0;}
+      int main(int argc, char *argv[]){return 0;}" 
+      C_HAS_${KEYWORD} )
+    set (CMAKE_REQUIRED_DEFINITIONS)
+
+    if (C_HAS_${KEYWORD})
+      set (C_INLINE TRUE)
+      if (NOT C_HAS_inline)
+        set (inline ${KEYWORD})
+      endif ()
+    endif (C_HAS_${KEYWORD})
+  endif (NOT DEFINED C_INLINE)
+endforeach (KEYWORD)
+
+configure_file (${CMAKE_CURRENT_SOURCE_DIR}/.cmake/config.h.in ${CMAKE_CURRENT_SOURCE_DIR}/config.h)
+configure_file (${CMAKE_CURRENT_SOURCE_DIR}/.cmake/cypher-parser.h.in ${CMAKE_CURRENT_SOURCE_DIR}/lib/src/cypher-parser.h)
+
+set (MODULE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/.cmake/Modules")
+set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${MODULE_DIR})
+
+check_c_compiler_flag ("-fPIC" CC_HAVE_FPIC)
+check_c_compiler_flag ("-fvisibility=hidden" CC_HAVE_VISIBILITY)
+
+# Linker flags are ignored by check_c_compiler_flag, so we have
+# to use this workaround
+set (CMAKE_REQUIRED_FLAGS "-Wl,--exclude-libs=ALL")
+check_c_compiler_flag ("" LINKER_HAS_EXCLUDE_LIBS)
+unset (CMAKE_REQUIRED_FLAGS)
+
+if (LINKER_HAS_EXCLUDE_LIBS)
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wl,--exclude-libs=ALL")
+endif (LINKER_HAS_EXCLUDE_LIBS)
+
+if (CC_HAVE_FPIC)
+  set (PIC_C_FLAGS "-fPIC")
+endif (CC_HAVE_FPIC)
+
+if (CC_HAVE_VISIBILITY)
+  set (VISI_C_FLAGS "-fvisibility=hidden")
+endif (CC_HAVE_VISIBILITY)
+
+function (cr_link_libraries _TARGET)
+  set (multiValueArgs IF)
+  cmake_parse_arguments (ARGS "" "" "${multiValueArgs}" ${ARGN})
+  if (ARGS_IF)
+    if (${ARGS_IF})
+    else (${ARGS_IF})
+      return ()
+    endif (${ARGS_IF})
+  endif (ARGS_IF)
+  target_link_libraries (${_TARGET} ${ARGS_UNPARSED_ARGUMENTS})
+endfunction (cr_link_libraries)
+
+find_package (check)
+find_package (fmem)
+find_package (leg)
+
+list (APPEND CMAKE_REQUIRED_DEFINITIONS
+  -D_GNU_SOURCE
+  -D_CRT_RAND_S
+  -DVC_EXTRALEAN
+  -DWIN32_LEAN_AND_MEAN)
+
+set (SOURCES)
+
+list (APPEND SOURCES 
+    lib/src/ast.c
```

Comment:
This list duplicates the list in `lib/src/Makefile.am`. Is there any way to avoid that duplication?

Reply:
Yes, I will add a macro in `CMakeLists.txt` to parse the source files list out of `lib/src/Makefile.am`.